### PR TITLE
Contract deployer fixes

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

Various changes are required to the contract deployer to support preallocs.

### What changes were made as part of this PR:

Functional.

- Logs better error messages
- Makes sure prealloc tx succeeds
- Proper handling of default constructor params

### What are the key areas to look at

- ...

### :rotating_light: Definition of Done :rotating_light:
- [ ] Deployed into dev-testnet 
- [ ] Passes tests on dev-testnet
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: tools/contractdeployer/cli.go [hunk 1]
```diff
@@ -6,6 +6,11 @@ import (
 	"strings"
 )
 
+const (
+	chainIDPlaceholder           = 0
```
Comment (BedrockSquirrel) [inline tools/contractdeployer/cli.go:10]: Little comment here might be good to say these are the go default values used for comparison to see if an argument value was passed.

File: tools/contractdeployer/contract_deployer.go [hunk 2]
```diff
@@ -203,25 +206,22 @@ func signAndSendTxWithReceipt(wallet wallet.Wallet, client ethadapter.EthClient,
 	}
 
 	var start time.Time
-	var receipt *types.Receipt
-	var contractAddr *common.Address
 	for start = time.Now(); time.Since(start) < timeoutWait; time.Sleep(retryInterval) {
-		receipt, err = client.TransactionReceipt(signedTx.Hash())
+		receipt, err := client.TransactionReceipt(signedTx.Hash())
+		if err != nil {
+			log.Info(err.Error())
```
Comment (BedrockSquirrel) [inline tools/contractdeployer/contract_deployer.go:212]: Maybe add something to this like `"Attempt failed, retrying... %w"` so if anyone does look at these logs they don't get too worried by the semi-spam

File: tools/contractdeployer/contract_deployer.go [hunk 3]
```diff
@@ -203,25 +206,22 @@ func signAndSendTxWithReceipt(wallet wallet.Wallet, client ethadapter.EthClient,
...
+			return &receipt.ContractAddress, nil
 		}
 
 		log.Info("Contract deploy tx has not been mined into a block after %s...", time.Since(start))
 	}
-	if contractAddr == nil {
-		return nil, fmt.Errorf("failed to mine contract deploy tx %s into a block after %s. Aborting", signedTx.Hash(), time.Since(start))
-	}
-	return contractAddr, nil
+	return nil, fmt.Errorf("failed to mine contract deploy tx %s into a block after %s. Aborting", signedTx.Hash(), time.Since(start))
```
Comment (BedrockSquirrel) [inline tools/contractdeployer/contract_deployer.go:224]: I think the idiomatic Go style tries to keep the happy path as the final return but this is a weird one and I don't feel strongly.

**Review by BedrockSquirrel — COMMENTED** [review]
few thoughts

File: tools/contractdeployer/cli.go (hunk 1 above)
Comment (joeldudley) [reply tools/contractdeployer/cli.go:11]: Added.

File: tools/contractdeployer/contract_deployer.go (hunk 2 above)
Comment (joeldudley) [reply tools/contractdeployer/contract_deployer.go]: Left in by accident. Removed.

File: tools/contractdeployer/contract_deployer.go (hunk 3 above)
Comment (joeldudley) [reply tools/contractdeployer/contract_deployer.go:221]: Cool. I'll leave it for now

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**